persistence/scaladsl/src/test/scala/com/lightbend/lagom/scaladsl/persistence/TestEntity.scala

```diff
@@ -67,10 +66,59 @@ object TestEntity {
   final case class AfterRecovery(state: State)
 }
 
-class TestEntity @Inject() (system: ActorSystem, probe: Option[ActorRef] = None)
+// NOTE needs to be registered in actor system config to be picked up like this:
+// lagom.serialization.play-json.serialization-registry =
+//   "com.lightbend.lagom.scaladsl.persistence.TestEntitySerializerRegistry"
+class TestEntitySerializerRegistry extends SerializerRegistry {
+  import TestEntity._
+  import play.api.libs.json._
+
+  implicit val modeFormat = Format[Mode](
+    Reads[Mode] {
+      case JsString("append")  => JsSuccess(Mode.Append)
+      case JsString("prepend") => JsSuccess(Mode.Prepend)
+      case js                  => JsError(s"unknown mode js: $js")
+    }, Writes[Mode] {
+      case Mode.Append  => JsString("append")
+      case Mode.Prepend => JsString("prepend")
+    }
+  )
+
+  def emptySingletonFormat[A](singleton: A) = Format[A](
+    Reads[A](_ => JsSuccess(singleton)),
+    Writes[A](_ => JsObject(Seq.empty))
+  )
+
+  override def serializers: Seq[Serializers[_]] = Seq(
+    // commands
+    Serializers(Json.format[Add]),
+    Serializers(Json.format[ChangeMode]),
+    Serializers(emptySingletonFormat(Get)),
+    Serializers(emptySingletonFormat(UndefinedCmd)),
+    Serializers(emptySingletonFormat(GetAddress)),
+
+    // events
+    Serializers(Json.format[Appended]),
+    Serializers(Json.format[Prepended]),
+    Serializers(emptySingletonFormat(InPrependMode)),
+    Serializers(emptySingletonFormat(InAppendMode)),
+
+    // state
+    Serializers(Json.format[State])
+  )
```

Comment:
This highlights one problem with using play-json for this, we need to call the macros for the concrete types to get a separate serializer per type, that means we cannot match on the sealed trait and therefore will get errors about missing serializers at runtime rather than compile time.

One option would be to manually construct a format for `Cmd` and one for `Evt` but then we'd have to provide our own logic to encode and decode which concrete subtype the serialized data is and I think we also would have to provide two partial functions, one for reads and one for writes, so that is a lot of boilerplate.